### PR TITLE
[MB-14323, MB-14296, MB-14208] audit history table migrations

### PR DESCRIPTION
**Pull request by mmayo-truss**
## Summary
This PR is grouping the migrations together for multiple issues. 

MB-14323: We are adding to the ignored_cols array for multiple to remove duplicate entries on the move history table when a new item is added. 
Note that this will remove these ids from being tracked in the `changed_values` column of the audit history table. There are some foreign keys that need to be tracked, mostly with duty locations, so I omitted those from the ignored columns and commented that in the migration file. 
Second Note: this migration file looks bigger than it is. I decided to change the formatting of how we typically use the `add_audit_history_table` function in order to make it easier to see all of the ignored columns and make appropriate comments.

MB-14208: We are adding a the `user_uploads` table to be tracked on the audit history table. 

MB-14296: We are adding the `backup_contacts` table to be tracked on the audit history table.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use two separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the milmove app as a service member.
2. Go through the process of creating a move (Make sure you use the correct GBLOC, Fort McCoy is a duty station that will show up easily for our normal local dev patterns)
3. When you create the shipment do not add a secondary pickup or a secondary delivery address
4. 🛑 Don't submit the move yet
5. Go back and edit address to add a secondary pickup address
6. 🛑 Don't submit the move yet
7. Go back again to edit address to add a secondary delivery address
8. Sign and submit the move.
9. Log into the office app as a services counselor and view the move history for the move just created.
10. You should see the secondary addresses displayed as their own rows on the move history table, but not any accompanying empty rows for the id being attached to the shipment.
11. You will NOT see any entries currently for the `user_uploads` or the `backup_contacts`, since the query has not been modified. But if you are running this locally you should see entries on a db viewer for both the orders upload and the creation of the backup contacts when you went through the creation of the move.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Have the Jira acceptance criteria been met for this change?

## Screenshots
BEFORE:
image

AFTER
image

DB view of audit history table featuring backup_contacts and user_upload entries:
image

### CODE REVIEW

**Review by juliemoon — APPROVED** [review]
@mmayo-truss this looks great! Thanks for getting this done!
Screen Shot 2022-10-21 at 2 54 54 PM

**Review by afoon — APPROVED** [review]
Screen Shot 2022-10-21 at 2 50 11 PM
LGTM! Look at how much space was saved with the filtering 🥹.

**Review by kctruss — APPROVED** [review]
Looks good! I tested the migration on EXP and everything worked beautifully. Well done!

File: migrations/app/schema/20221020202612_create-and-modify-move-history-triggers-for-multiple-tables.up.sql [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	audit_rows := BOOLEAN 't',
+	audit_query_text := BOOLEAN 't',
+	ignored_cols := ARRAY[
+		'created_at',
+		'updated_at',
+		'service_member_id',
+		'uploaded_orders_id',
+		'entitlement_id',
+		'uploaded_amended_orders_id'
+	] -- origin_duty_location_id and new_duty_location_id are fks but are utilized to display supplemental information
```
Comment (afoon) [inline migrations/app/schema/20221020202612_create-and-modify-move-history-triggers-for-multiple-tables.up.sql:85]: Should `rank` be omitted here too? It's also part of the `service_member` table and results on duplicate entries if rank is updated.
Screen Shot 2022-10-21 at 6 09 55 PM

Comment (mmayo-truss) [reply migrations/app/schema/20221020202612_create-and-modify-move-history-triggers-for-multiple-tables.up.sql:85]: @afoon we can totally omit rank if we do not want to show the duplicate entries here. I was just focusing on foreign key ids, but this makes sense.

**Comment by mmayo-truss** [comment]
I added `grade` (which correlates to `rank`) to the ignored cols for the `orders` table to hide the duplicated rank rows as seen in this PR #9295. I also updated the event names to "Updated profile" as per @juliemoon's instructions. 
image

**Review by kctruss — APPROVED** [review]
Approved (again) after new changes pushed. Thanks for all of this @mmayo-truss !

**Comment by juliemoon** [comment]
Thanks @mmayo-truss I see that the rank is not getting duplicated and that the event name when a customer updates their rank shows up as **Updated profile**.
Screen Shot 2022-10-24 at 12 59 40 PM

**Pull request merged**